dune/stuff/functions/expression/mathexpr.cc

```diff
@@ -21,336 +21,620 @@ This software comes with absolutely no warranty.
 {
-  type=1;name=new char[1];name[0]=0;
-  nvars=nvarsp;
-  if(nvars){
-    ppvar=new PRVar[nvars];
-    int i;for(i=0;i<nvars;i++)ppvar[i]=ppvarp[i];
-    buf=new double[nvars];
-  }else {ppvar=NULL;buf=NULL;}
+  type = 1;
+  name = new char[1];
+  name[0] = 0;
+  nvars = nvarsp;
+  if (nvars) {
+    ppvar = new PRVar[nvars];
+    int i;
+    for (i = 0; i < nvars; i++)
+      ppvar[i] = ppvarp[i];
+    buf = new double[nvars];
+  } else {
+    ppvar = NULL;
+    buf = NULL;
+  }
 }
 
 RFunction::~RFunction()
 {
-  if(name!=NULL)delete[]name;
-  if(ppvar!=NULL)delete[]ppvar;
-  if(buf!=NULL)delete[]buf;
+  if (name != NULL)
+    delete[] name;
+  if (ppvar != NULL)
+    delete[] ppvar;
+  if (buf != NULL)
+    delete[] buf;
 }
 
 RFunction& RFunction::operator=(const RFunction& rfunc)
 {
-  if(this==&rfunc)return *this;
-  type=rfunc.type;op=rfunc.op;
-  pfuncval=rfunc.pfuncval;
-  delete[]name;
-  name=CopyStr(rfunc.name);
-  if(ppvar!=NULL)delete[]ppvar;
-  ppvar=NULL;
-  if(buf!=NULL)delete[]buf;
-  buf=NULL;
-  nvars=rfunc.nvars;
-  if(type==1&&nvars){
-    ppvar=new PRVar[nvars];buf=new double[nvars];
-    int i;for(i=0;i<nvars;i++)ppvar[i]=rfunc.ppvar[i];
+  if (this == &rfunc)
+    return *this;
+  type = rfunc.type;
+  op = rfunc.op;
+  pfuncval = rfunc.pfuncval;
+  delete[] name;
+  name = CopyStr(rfunc.name);
+  if (ppvar != NULL)
+    delete[] ppvar;
+  ppvar = NULL;
+  if (buf != NULL)
+    delete[] buf;
+  buf = NULL;
+  nvars = rfunc.nvars;
+  if (type == 1 && nvars) {
+    ppvar = new PRVar[nvars];
+    buf = new double[nvars];
+    int i;
+    for (i = 0; i < nvars; i++)
+      ppvar[i] = rfunc.ppvar[i];
   }
   return *this;
 }
 
-void RFunction::SetName(const char*s)
-{if(name!=NULL)delete[]name;name=CopyStr(s);}
+void RFunction::SetName(const char* s)
+{
+  if (name != NULL)
+    delete[] name;
+  name = CopyStr(s);
+}
 
 double RFunction::Val(double x) const
 {
-  if(type==-1||nvars>=2)return ErrVal;
-  if(type==0)return (*pfuncval)(x);
-  double xb=*(*ppvar)->pval,y;
-  *(*ppvar)->pval=x;  // Warning : could cause trouble if this value is used in a parallel process
-  y=op.Val();
-  *(*ppvar)->pval=xb;
+  if (type == -1 || nvars >= 2)
+    return ErrVal;
+  if (type == 0)
+    return (*pfuncval)(x);
+  double xb = *(*ppvar)->pval, y;
+  *(*ppvar)->pval = x; // Warning : could cause trouble if this value is used in a parallel process
+  y = op.Val();
+  *(*ppvar)->pval = xb;
   return y;
 }
 
-double RFunction::Val(double*pv) const
+double RFunction::Val(double* pv) const
 {
-  if(type==-1)return ErrVal;
-  if(type==0)return (*pfuncval)(*pv);
+  if (type == -1)
+    return ErrVal;
+  if (type == 0)
+    return (*pfuncval)(*pv);
   double y;
   int i;
-  for(i=0;i<nvars;i++){
-    buf[i]=*ppvar[i]->pval;
+  for (i = 0; i < nvars; i++) {
+    buf[i] = *ppvar[i]->pval;
     // Warning : could cause trouble if this value is used in a parallel process
```

Comment:
Totally unrelated: I just read this. I would really like to get rid of this code... *sigh*